File: templates/service-java/{{#server.spring}}/src/main/java/{{package.value}}/{{#jsonlib.moshi}}/MoshiConfig.java
package {{package.value}};

import com.squareup.moshi.Moshi;
import org.springframework.context.annotation.*;
import {{package.value}}.json.*;

@Configuration
public class MoshiConfig {
	@Bean
	@Primary
	public Moshi getMoshi() {
		var moshiBuilder = new Moshi.Builder();
		CustomMoshiAdapters.setup(moshiBuilder);
		return moshiBuilder.build();
	}

    @Bean
    public Json json() {
        return new Json(getMoshi());
    }
}

File: templates/service-kotlin/{{#server.spring}}/src/main/kotlin/{{package.value}}/{{#jsonlib.moshi}}/MoshiConfig.kt
package {{package.value}}

import com.squareup.moshi.Moshi
import org.springframework.context.annotation.*
import {{package.value}}.json.*

@Configuration
class MoshiConfig {
    @Bean
    @Primary
    fun getMoshi(): Moshi {
        val moshiBuilder = Moshi.Builder()
        setupMoshiAdapters(moshiBuilder)
        return moshiBuilder.build()
    }

    @Bean
    fun json(): Json {
        return Json(getMoshi())
    }
}